pkg/cluster/mpi/mpi.go

```diff
@@ -1,33 +1,37 @@
 package mpi
 
 import (
-	"path/filepath"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 	"time"
 
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
 	dcv1alpha1 "github.com/dominodatalab/distributed-compute-operator/api/v1alpha1"
 	"github.com/dominodatalab/distributed-compute-operator/pkg/cluster/metadata"
 )
 
 const (
-	sshdPort                  int32 = 2222
-	sshdPortName                    = "sshd"
-	sshVolumeName                   = "ssh-auth"
-	sshRootMountPath                = "/etc/ssh"
-	sshAuthorizedKeysFilename       = "authorized_keys"
-	sshAuthPublicKey                = "ssh-publickey"
 
-	configVolumeName    = "config"
-	configRootMountPath = "/etc/mpi"
+	// SSH port used by MPI and a name of this port within the service
+	sshdPort     = 2222
+	sshdPortName = "sshd"
 
-	hostFileFilename   = "hostfile"
-	sshdConfigFilename = "sshd_config"
+	// Locations of the mounted files and their modes
+	authorizedKeysPath = "/etc/mpi/authorized_keys"
+	authorizedKeysMode = 0444 // octal!
+	launchScriptPath   = "/opt/domino/bin/mpi-worker-start.sh"
+	launchScriptMode   = 0544 // octal!
 
-	finalizerRetryPeriod = 1 * time.Second
-)
+	// Default parameters of a user account for executing MPI workload.
+	defaultUserID    = 12574
+	defaultUserName  = "domino"
+	defaultGroupID   = 12574
+	defaultGroupName = "domino"
```

Comment:
Do we want to call this 'mpi' and then override it via the CR? Just to maintain some semblance of this as an open source product?

Reply:
I wasn't sure what the right name should be. The main definition is this: https://github.com/dominodatalab/distributed-compute-operator/blob/77c4378769ac3ace4b078054c41399f42212258b/api/v1alpha1/mpicluster_types.go#L13. The names of these constants should derive from that.

Reply:
Let me know if you have suggestions for better names (for CR fields). Now is a good time to change this.